jboss/container/eap/prometheus/config/6.4/artifacts/opt/jboss/container/prometheus/etc/jmx-exporter-config.yaml

```diff
@@ -0,0 +1,354 @@
+# limitations under the License.
+#
+---
+lowercaseOutputName: true
+lowercaseOutputLabelNames: true
+whitelistObjectNames:
+- "jboss.as:deployment=*,subsystem=ejb3,*"
+- "jboss.as:deployment=*,subsystem=web"
+- "jboss.as:deployment=*,subsystem=web,servlet=*"
+- "jboss.as:deployment=*,subdeployment=*,subsystem=ejb3,*"
+- "jboss.as:deployment=*,subdeployment=*,subsystem=web"
+- "jboss.as:deployment=*,subdeployment=*,subsystem=web,servlet=*"
+- "jboss.as:subsystem=datasources,data-source=*,statistics=jdbc"
+- "jboss.as:subsystem=datasources,data-source=*,statistics=pool"
+- "jboss.as:subsystem=datasources,xa-data-source=*,statistics=jdbc"
+- "jboss.as:subsystem=datasources,xa-data-source=*,statistics=pool"
+- "jboss.as:subsystem=messaging,hornetq-server=*,jms-queue=*"
+- "jboss.as:subsystem=messaging,hornetq-server=*,jms-topic=*"
+- "jboss.as:subsystem=transactions"
+blacklistObjectNames:
+# handled by agent's default exporter
+- "java.lang:*"
+rules:
+# EAP Specific
+# Web Metrics
+# Sessions
+- pattern: "jboss.as<deployment=(.+), *subsystem=web><>(active_sessions|max_active_sessions): (.+)"
+  name: web_sessions_$2
+  help: Web Session Metrics $2
+  value: $3
+  valueFactor: 1
+  labels:
+    deployment: $1
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<deployment=(.+), *subdeployment=(.+), *subsystem=web><>(active_sessions|max_active_sessions): (.+)"
+  name: web_sessions_$3
+  help: Web Session Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    deployment: $1
+    subdeployment: $2
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<deployment=(.+), *subsystem=web><>(session_avg_alive_time|session_max_alive_time): (.+)"
+  name: web_sessions_$2_seconds
+  help: Web Session Metrics $2
+  value: $3
+  valueFactor: .001
+  labels:
+    deployment: $1
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<deployment=(.+), *subdeployment=(.+), *subsystem=web><>(session_avg_alive_time|session_max_alive_time): (.+)"
+  name: web_sessions_$3_seconds
+  help: Web Session Metrics $3
+  value: $4
+  valueFactor: .001
+  labels:
+    deployment: $1
+    subdeployment: $2
+  attrNameSnakeCase: true
+  type: GAUGE
+- pattern: "jboss.as<deployment=(.+), *subsystem=web><>(duplicated_session_ids|expired_sessions|rejected_sessions|sessions_created): (.+)"
+  name: web_sessions_$2
+  help: Web Session Metrics $2
+  value: $3
+  valueFactor: 1
+  labels:
+    deployment: $1
+  attrNameSnakeCase: true
+  type: COUNTER
+- pattern: "jboss.as<deployment=(.+), *subdeployment=(.+), *subsystem=web><>(duplicated_session_ids|expired_sessions|rejected_sessions|sessions_created): (.+)"
+  name: web_sessions_$3
+  help: Web Session Metrics $3
+  value: $4
+  valueFactor: 1
+  labels:
+    deployment: $1
+    subdeployment: $2
+  attrNameSnakeCase: true
+  type: COUNTER
+# Servlets
+- pattern: "jboss.as<deployment=(.+), *subsystem=web, *servlet=(.+)><>(load_time|max_time|min_time): (.+)"
+  name: web_servlets_$3_seconds
```

Comment:
Should the name be unique?

Affect, the name is composed of the attribute name only (`load_time`, 'max_time` or `min_time`).
How would you distinguish the metric `web_servlets_max_time_seconds` between different servlet (in different deployment)?

The Microprofile Metrics mandates that the metric name is unique in a given scope. So I ended up with a metric name composed of the resource address appended with the attribute name.

Reply:
It sounds like we have a conflict between the spec requirement and the "prometheusy" thing @rcernich mentioned above. I get the point of the "prometheusy" approach; it's more flexible.

Reply:
I agree that the MP spec is too restrictive (uniqueness should be checked with name + tags).

I'll raise the issue on https://github.com/eclipse/microprofile-metrics